src/actions/blogActions.js

```diff
@@ -1,72 +1,74 @@
-import { v4 as uuidv4 } from "uuid";
 import {
-  ADD_BLOG_POST,
-  EDIT_BLOG_POST,
   GET_BLOG_POSTS,
   SEARCH_BLOG_POSTS,
   FILTER_BY_TAG,
 } from "@constants/actionTypes";
 
-export const addBlog = (post) => {
-  return (dispatch) => {
-    try {
-      const id = uuidv4();
-      const createdAt = new Date().toLocaleDateString("en-US", {
-        month: "long",
-        day: "2-digit",
-        year: "numeric",
-      });
-      const postWithId = {
-        ...post,
-        id,
-        createdAt,
-      };
-
-      dispatch({
-        type: ADD_BLOG_POST,
-        payload: postWithId,
-      });
+import supabase from "./../app/supabase";
 
-      const existingPosts = JSON.parse(localStorage.getItem("blogPosts")) || [];
-      const updatedPosts = [postWithId, ...existingPosts];
-      localStorage.setItem("blogPosts", JSON.stringify(updatedPosts));
+export const fetchBlogs = () => {
+  return async (dispatch) => {
+    try {
+      const { data, error } = await supabase
+        .from("blogs")
+        .select("*")
+        .order("createdAt", { ascending: false });
+      if (error) {
+        throw new Error(error.message);
+      }
+      dispatch(getBlogs(data));
     } catch (error) {
-      throw new Error("Error adding blog post: " + error.message);
+      throw error;
     }
   };
 };
 
-export const getBlogPostsFromLocalStorage = () => {
-  return (dispatch) => {
+export const createBlogPost = (blog) => {
+  const newBlog = {
+    title: blog.title,
+    body: blog.body,
+    image: blog.image,
+    tags: blog.tags,
+    userId: blog.userId,
+    creatorImage: blog.creatorImage,
+    creatorFullName: blog.creatorFullName,
+  };
```

Comment:
if `blog` and `newBlog` has 1-1 mapping, then we can directly use `blog`
```suggestion
```

Reply:
got it vaiya.

Reply:
fix this . there is an extra proTypes in my code thats why i faced the issues on 1-1 mapping now resolved